### PR TITLE
Free Commutative Monoid

### PR DESCRIPTION
**Pull request by guilhermehas**
I added Free Commutative Monoid in the library.
To prove that it is isomorphic to the associative list, I had to add some auxiliary functions there.

### CODE REVIEW

**Review by mortberg — APPROVED** [review]
Very nice! Thanks for the contribution

File: Cubical/HITs/FreeComMonoids/Properties.agda [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+FCM→AL∘AL→FCM≡id = AL.ElimProp.f (trunc _ _) refl
+  λ x n {xs} p → ^-id x n (AL→FCM xs) ∙ cong (⟨ _ , _ ⟩∷_) p
+
+FCMon≃AssocList : AssocList A ≃ FreeComMonoid A
+FCMon≃AssocList = isoToEquiv (iso AL→FCM FCM→AL AL→FCM∘FCM→AL≡id FCM→AL∘AL→FCM≡id)
+
+AssocList≃FCMon : FreeComMonoid A ≃ AssocList A
+AssocList≃FCMon = isoToEquiv (iso FCM→AL AL→FCM FCM→AL∘AL→FCM≡id AL→FCM∘FCM→AL≡id)
+
+FCMon≡AssocList : AssocList A ≡ FreeComMonoid A
```
Comment (mortberg) [inline Cubical/HITs/FreeComMonoids/Properties.agda:95]: Wait, this name is a bit weird... It should be `AssocList≡FreeComMonoid`. Similarly for the two definitions above.

Comment (mortberg) [reply Cubical/HITs/FreeComMonoids/Properties.agda]: Great, can you also fix `FCMon≃AssocList`? And remove `AssocList≃FCMon`? I think it's unnecessary to have both when we can just use `invEquiv` to turn one of them around

Comment (guilhermehas) [reply Cubical/HITs/FreeComMonoids/Properties.agda]: I have already made these changes.

**Pull request merged**